build-utils/language-c/src/main/java/org/apache/plc4x/language/c/CLanguageTemplateHelper.java

```diff
@@ -380,7 +390,7 @@ public String getReadBufferReadMethodCall(SimpleTypeReference simpleTypeReferenc
                 throw new FreemarkerException("Unsupported float type with " + floatTypeReference.getSizeInBits() + " bits");
             case STRING:
                 StringTypeReference stringTypeReference = (StringTypeReference) simpleTypeReference;
-                return "plc4c_spi_read_string(io, " + stringTypeReference.getSizeInBits() + ", \"" +
+                return "plc4c_spi_read_string(io, " + stringTypeReference.getLength() + " * 8, \"" +
```

Comment:
What is the reason using bytes sized lengths here? Isn't that supported for strings?

Reply:
I added a getLength (now getLengthExpression) to the StringTypeReference class as it now uses an expression instead of a hard coded value.
getLengthExpression returns the size in bits, I have removed the "* 8"